etc/run.sh

```diff
@@ -0,0 +1,243 @@
+#!/bin/bash
+#source ~/.bashrc
+set -eou pipefail
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+declare _mail_d=~/mail
+
+_env_common() {
+    export PYKERN_PKDEBUG_WANT_PID_TIME=1
+}
+
+_env_mail_common() {
+    export SIREPO_SMTP_FROM_NAME=$USER+support@localhost.localdomain
+    export SIREPO_SMTP_FROM_EMAIL=DevSupport
+    if [[ ! ${SIREPO_AUTH_METHODS:-} =~ email ]]; then
+        export SIREPO_AUTH_METHODS=email:guest
+    fi
+}
+
+_env_mail_smtp() {
+    _env_mail_common
+    export SIREPO_SMTP_FROM_EMAIL=$USER+support@localhost.localdomainn
+    export SIREPO_SMTP_SEND_DIRECTLY=1
+    export SIREPO_SMTP_SERVER=localhost
+}
+
+_env_moderate() {
+    declare sim_type=$1
+    export SIREPO_FEATURE_CONFIG_MODERATED_SIM_TYPES=$sim_type
+    export SIREPO_AUTH_ROLE_MODERATION_MODERATOR_EMAIL=$USER+moderator@localhost.localdomain
+    _msg "Moderated sym_type=$sim_type"
+    _setup_smtp
+    _env_common
+}
+
+_err() {
+    _msg "$@"
+    return 1
+}
+
+_exec_all() {
+    _env_common
+    exec sirepo service http
+}
+
+_main() {
+    declare mode=${1:-missing-arg}
+    shift || true
+    declare f=_op_$mode
+    if [[ $(type -t "$f") != 'function' ]]; then
+        _err "invalid mode=$mode
+usage: bash ${BASH_SOURCE[0]} mode
+where mode is one of:
+$(compgen -A function _op_ | sed -e 's/^_op_//')"
+    fi
+    "$f" "$@"
+}
+
+_msg() {
+    echo "$@" 1>&2
+}
+
+_op_bluesky() {
+    export SIREPO_AUTH_BLUESKY_SECRET=bluesky-secret
+    export SIREPO_AUTH_METHODS=email:bluesky
+    _msg "To test, you need a sim, but this is the structure:
+
+curl -H 'Content-Type: application/json' -D - -X POST http://localhost:8000/auth-bluesky-login -d '{\"simulationId\": \"kRfyDC2q\", \"simulationType\": \"srw\"}'
+"
+    _op_mail
+}
+
+_op_flash() {
+    export SIREPO_FEATURE_CONFIG_PROPRIETARY_OAUTH_SIM_TYPES=flash
+    export SIREPO_SIM_OAUTH_FLASH_INFO_VALID_USER=G
+    if [[ ! ${SIREPO_SIM_OAUTH_FLASH_KEY:-} || ! ${SIREPO_SIM_OAUTH_FLASH_SECRET:-} ]]; then
+        echo 'You must set $SIREPO_SIM_OAUTH_FLASH_KEY and $SIREPO_SIM_OAUTH_FLASH_SECRET' 1>&2
+        exit 1
+    fi
+    _exec_all
+}
+
+_op_jupyterhub() {
+    # POSIT: versions same in container-beamsim-jupyter/build.sh
+    # Order is important: jupyterlab-server should be last so it isn't
+    # overwritten with a newer version.
+    declare p=$(pip freeze)
+    declare -a i=()
+    declare f
+    for f in \
+        jupyterhub==1.4.2 \
+        jupyterlab==3.1.14  \
+        notebook==6.5.4 \
+        jupyterlab_server==2.8.2\
```

Comment:
This should be jupyterlab-server (`-` not `_`)

Reply:
7cac2f975